webrtc-c/canary/src/CanarySignaling.cpp

```diff
@@ -1,6 +1,522 @@
+
+    // Create the lock and the cvar for iteration
+    CHK(IS_VALID_MUTEX_VALUE(lock = MUTEX_CREATE(FALSE)), STATUS_NOT_ENOUGH_MEMORY);
+    CHK(IS_VALID_CVAR_VALUE(terminateCv = CVAR_CREATE()), STATUS_NOT_ENOUGH_MEMORY);
+    CHK(IS_VALID_MUTEX_VALUE(canarySessionInfo.roundtripLock = MUTEX_CREATE(FALSE)), STATUS_NOT_ENOUGH_MEMORY);
+    CHK(IS_VALID_CVAR_VALUE(canarySessionInfo.roundtripCv = CVAR_CREATE()), STATUS_NOT_ENOUGH_MEMORY);
+    canarySessionInfo.exitStatus = STATUS_SUCCESS;
+    canarySessionInfo.iterationFailCount = 0;
+
+    CHK_STATUS(Canary::Cloudwatch::init(pConfig));
+    CHK_STATUS(initKvsWebRtc());
+    initialized = TRUE;
+
+    SET_LOGGER_LOG_LEVEL(pConfig->logLevel);
+    pConfig->print();
+
+    // The timer loop for iteration
+    CHK_STATUS(timerQueueCreate(&timerQueueHandle));
+
+    // We will create a static credential provider. We can replace it with others if needed.
+    CHK_STATUS(createStaticCredentialProvider((PCHAR) pConfig->pAccessKey, 0,
+                                              (PCHAR) pConfig->pSecretKey, 0,
+                                              (PCHAR) pConfig->pSessionToken, 0,
+                                              MAX_UINT64,
+                                              &pCredentialProvider));
+
+    // Generate a random channel name
+    generateChannelName(channelName);
+
+    // Prepare the channel info structure
+    masterChannelInfo.version = CHANNEL_INFO_CURRENT_VERSION;
+    masterChannelInfo.pChannelName = channelName;
+    masterChannelInfo.pRegion = (PCHAR) pConfig->pRegion;
+    masterChannelInfo.pKmsKeyId = NULL;
+    masterChannelInfo.tagCount = 0;
+    masterChannelInfo.pTags = NULL;
+    masterChannelInfo.channelType = SIGNALING_CHANNEL_TYPE_SINGLE_MASTER;
+    masterChannelInfo.channelRoleType = SIGNALING_CHANNEL_ROLE_TYPE_MASTER;
+    masterChannelInfo.cachingPolicy = SIGNALING_API_CALL_CACHE_TYPE_FILE;
+    masterChannelInfo.cachingPeriod = SIGNALING_API_CALL_CACHE_TTL_SENTINEL_VALUE;
+    masterChannelInfo.asyncIceServerConfig = TRUE;
+    masterChannelInfo.retry = TRUE;
+    masterChannelInfo.reconnect = TRUE;
+    masterChannelInfo.pCertPath = NULL; // Will be set by lookForCaCert
+    masterChannelInfo.messageTtl = 0; // Default is 60 seconds
+
+    masterSignalingClientCallbacks.version = SIGNALING_CLIENT_CALLBACKS_CURRENT_VERSION;
+    masterSignalingClientCallbacks.errorReportFn = signalingClientError;
+    masterSignalingClientCallbacks.stateChangeFn = signalingClientStateChanged;
+    masterSignalingClientCallbacks.messageReceivedFn = signalingMessageReceived;
+    masterSignalingClientCallbacks.customData = (UINT64) &canarySessionInfo;
+
+    masterClientInfo.version = SIGNALING_CLIENT_INFO_CURRENT_VERSION;
+    masterClientInfo.loggingLevel = pConfig->logLevel;
+    STRCPY(masterClientInfo.clientId, SIGNALING_CANARY_MASTER_CLIENT_ID);
+
+    // Populate the ca cert
+    lookForCaCert(&masterChannelInfo);
```

Comment:
No need to lookForCaCert, you can just pass `DEFAULT_KVS_CACERT_PATH`. It's set in the cmake level, https://github.com/aws-samples/amazon-kinesis-video-streams-demos/blob/8bf1976fa7d53b54335bfaba56b9266d32b82a48/webrtc-c/canary/CMakeLists.txt#L38.

Reply:
Aaah! Let me get rid of this all